holoviews/core/data/pandas.py

```diff
@@ -80,6 +81,16 @@ def init(cls, eltype, data, kdims, vdims):
                 if eltype._auto_indexable_1d and ncols == 1 and kdim not in data.columns:
                     data = data.copy()
                     data.insert(0, kdim, np.arange(len(data)))
+
+            col_counter = Counter(data.columns)
+            for d in kdims+vdims:
+                if isinstance(d, Dimension): d = d.name
+                count = col_counter[d]
+                if d in col_counter and count > 1:
+                    raise DataError('Dimensions may not reference duplicated DataFrame '
+                                    'columns (found %d %r columns). If you want to plot '
+                                    'a column against itself simply declare two dimensions '
+                                    'with the same name. '% (count, d), cls)
```

Comment:
I don't really see a need to use ``Counter`` for this. I think I would have done something like:

```python
columns = []
for d in kdims+vdims:
   d = d.name if isinstance(d, Dimension) else d
    if d in columns:
        raise DataError('Dimensions may not reference duplicated DataFrame '
                                     'column  %r. If you want to plot '
                                    'a column against itself simply declare two dimensions '
                                    'with the same name. '% d, cls)
   columns.append(d)
```

It is just as long as avoids an import.

Reply:
That logic is wrong, we're checking for referenced columns that are duplicated in the dataframe, not duplicate dimensions. Happy to rewrite it without ``Counter`` though.